templates/project/build.rs

```diff
@@ -1,277 +1,66 @@
-        rt.block_on(read_certificates(&mut cfg))?;
-        (robot_cfg, cfg)
-    } else {
-        (RobotConfig::default(), Config::default())
-    };
-
-    let cloud_cfg = robot_cfg.cloud.unwrap_or_default();
-    let robot_name = cloud_cfg.local_fqdn.split('.').next().unwrap_or("");
-    let local_fqdn = cloud_cfg.local_fqdn.replace('.', "-");
-    let fqdn = cloud_cfg.fqdn.replace('.', "-");
-
-    let mut certs = cfg
-        .cloud
-        .tls_certificate
-        .split_inclusive("----END CERTIFICATE-----");
-    let mut srv_cert = (&mut certs).take(2).collect::<String>();
-    srv_cert.push('\0');
-    let ca_cert = certs
-        .take(1)
-        .map(der::Document::from_pem)
-        .filter(|s| s.is_ok())
-        .map(|s| s.unwrap().1.to_vec())
-        .collect::<Vec<Vec<u8>>>()
-        .pop()
-        .unwrap_or_default();
-    let key = der::Document::from_pem(&cfg.cloud.tls_private_key)
-        .map_or(vec![], |k| k.1.as_bytes().to_vec());
-
-    let out_dir = env::var_os("OUT_DIR").unwrap();
-    let dest_path = Path::new(&out_dir).join("robot_secret.rs");
-    let robot_decl = vec![
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_ID = cfg.cloud.id.as_str()
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_SECRET = cfg.cloud.secret.as_str()
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            LOCAL_FQDN = local_fqdn.as_str()
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            FQDN = fqdn.as_str()
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_NAME = robot_name
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_DTLS_CERT = cert_der
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_DTLS_KEY_PAIR = kp_der
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_DTLS_CERT_FP = fp
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_SRV_PEM_CHAIN = srv_cert.as_bytes()
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_SRV_PEM_CA = ca_cert
-        ),
-        const_declaration!(
-            #[allow(clippy::redundant_static_lifetimes, dead_code)]
-            ROBOT_SRV_DER_KEY = key
-        ),
-    ]
-    .join("\n");
-    fs::write(dest_path, robot_decl).unwrap();
 
-    let metadata = MetadataCommand::new()
+    let metadata = cargo_metadata::MetadataCommand::new()
```

Comment:
I don't mind it, I'm just curious why you changed from `use cargo_metadata::{MetadataCommand, CargoOpt}` to long names for these types.

Reply:
I was lazy :P